davici.c

```diff
@@ -131,23 +104,21 @@ static void ip_and_port(char *ip_address, char *ip, char *port)
 	}
 }
 
-static int connect_in_and_fcntl(int fd, const char *path)
+static int connect_in_and_fcntl(int fd, const char *address)
 {
 	struct sockaddr_in addr;
 	int flags;
 
 	memset(&addr, 0, sizeof(addr));
 	addr.sin_family = AF_INET;
-	char *ip_address = malloc(strlen(path) * sizeof(char));
-	char *ip = malloc(strlen(path) * sizeof(char));
+	char *ip = malloc(strlen(address) * sizeof(char));
```

Comment:
this should be, likewise for others:

char *ip = malloc(strlen(address) + 1); 

To include the null characeter as strlen always returns the number of characters not including the null terminator, you technically don't need the sizeof(char) as it's always going to be 1 but can leave it for readability. 

I see the original code doesn't add the +1 which I think is a bug in their code

Reply:
Have gone with your example.